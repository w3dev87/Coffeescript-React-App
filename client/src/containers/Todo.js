import { connect } from 'react-redux';
import * as todoActions from '../actions/todoActions';
import Todo from '../components/Todo';

// map state from store to props
const mapStateToProps = (state) => {
  return {
    mappedTodoState: state.todoState
  }
}

// map actions to props
const mapDispatchToProps = (dispatch) => {
  return {
    mappedfetchTodoById: todoId => dispatch(todoActions.fetchTodoById(todoId))
  }
}

export default connect(mapStateToProps,mapDispatchToProps)(Todo);
