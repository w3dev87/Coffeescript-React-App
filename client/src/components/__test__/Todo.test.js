import React from 'react';
import Todo, * as toDoActions from '../Todo';
import { shallow } from 'enzyme';
describe('Todo', () => {
    let component = null;
    const id = 0;
    const fake_props = { 
        todo: {            
            todoText:'Study', 
            todoDesc:'English'
        }, 
        isFetching: true 
    };

    it('renders correctly', () => {
    component = shallow(<Todo 
        mappedfetchTodoById = {jest.fn()} 
        params = {id} 
        mappedTodoState = {fake_props}/>);
    });

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    })

});


