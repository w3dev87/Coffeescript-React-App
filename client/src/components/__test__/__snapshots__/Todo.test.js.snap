// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo matches snapshot 1`] = `
<div
  className="todoDetail"
>
  <h2>
    Todo Detail
  </h2>
</div>
`;
